src/main/java/edu/kis/powp/jobs2d/command/CommandTransformation.java

```diff
@@ -0,0 +1,51 @@
+package edu.kis.powp.jobs2d.command;
+
+import edu.kis.powp.jobs2d.Job2dDriver;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+public abstract class CommandTransformation implements CommandVisitorInterface {
+    List<DriverCommand> commands;
+
+    public CommandTransformation(){
+        commands = new ArrayList<>();
+    }
+
+    public ICompoundCommand getTransformedCommand() {
+        return new ICompoundCommand() {
+            @Override
+            public Iterator<DriverCommand> iterator() {
+                return commands.iterator();
+            }
+
+            @Override
+            public void execute(Job2dDriver driver) {
+                commands.forEach(c -> c.execute(driver));
+            }
+        };
```

Comment:
Are there no ICompoundCommand implementations to re-use?

Reply:
As far as I can see, there are no such implementations.

Reply:
I see at least two:
https://github.com/iis-powp-2020/powp_jobs2d_projects/tree/IO-3-dev/src/main/java/edu/kis/powp/jobs2d/command